iroh-embed/src/lib.rs

```diff
@@ -86,14 +92,50 @@ pub use store::RocksStoreService;
 ///                     .unwrap();
 /// # })
 /// ```
-#[derive(Debug)]
+///
+/// Using an in-memory store:
+/// ```no_run
+/// use iroh_embed::{Iroh, IrohBuilder, IrohService, Libp2pConfig, P2pService, MemStoreService};
+/// use testdir::testdir;
+/// # tokio_test::block_on(async {
+/// let store = MemStoreService::new().await.unwrap();
+/// let mut p2p_config = Libp2pConfig::default();
+/// p2p_config.listening_multiaddrs = vec![
+///     "/ip4/0.0.0.0/tcp/0".parse().unwrap(),  // random port
+///     "/ip4/0.0.0.0/udp/0/quic-v1".parse().unwrap(),  // random port
+/// ];
+/// let dir = testdir!();
+/// let p2p = P2pService::new(p2p_config, dir, store.addr()).await.unwrap();
+/// let _iroh: Iroh = IrohBuilder::new()
+///                     .store(store)
+///                     .p2p(p2p)
+///                     .build()
+///                     .await
+///                     .unwrap();
+/// # })
+/// ```
+// Note: The doc tests above are disabled because they run extremely slow.  However the same
+// tests run just fine as normal tests.  We should really figure out what is going on
+// sometime.
+
 pub struct IrohBuilder {
-    store: Option<RocksStoreService>,
+    store: Option<Box<dyn IrohService<StoreService>>>,
```

Comment:
Why `Box` over generic?

Reply:
Hum, because I never thought of a builder as a generic struct and imagined someone calling repeatedly `.store()` with different stores...  But I guess that's a bit of a useless usecase.  Generic would get rid of the weird `fn stop(self: Box<Self>)` thing I had to do so that'd be a plus.  I'll change it and see what it looks like.